pkg/policy/policy.go

```diff
@@ -80,17 +79,40 @@ func (s *SearchContext) String() string {
 	}
 	for _, dport := range s.DPorts {
 		if dport.Name != "" {
-			dports = append(dports, fmt.Sprintf("%s/%s", dport.Name, dport.Protocol))
+			var str strings.Builder
+			str.Grow(len(dport.Name) + len(dport.Protocol) + 1)
+			str.WriteString(dport.Name)
+			str.WriteRune('/')
+			str.WriteString(dport.Protocol)
+			dports = append(dports, str.String())
 		} else {
-			dports = append(dports, fmt.Sprintf("%d/%s", dport.Port, dport.Protocol))
+			var str strings.Builder
```

Comment:
Nit: `str` declaration can be moved before the if statement because it's created both branches